queue_job/controllers/main.py

```diff
@@ -87,10 +58,23 @@ def retry_postpone(job, message, seconds=None):
             job.store()
             env.cr.commit()
 
-        job = self._load_job(env, job_uuid)
-        if job is None:
+        # ensure the job to run is in the correct state and lock the record
+        env.cr.execute(
+            "SELECT state FROM queue_job "
+            "WHERE uuid=%s AND state=%s "
+            "FOR UPDATE",
```

Comment:
Do we want to use NOWAIT and catch concurrent errors, to avoid locking the second worker?

Reply:
@guewen I thought about that and opted for the simpler code. My thinking is the benefit is marginal (a nicer log in rare situations) for a more complex code. The lock will never be held long (ie until the job is started immediately after).

Reply:
Didn't realize it was only until it starts, better to keep it simple yes :)